soccer/planning/low_level/RRTUtil.cpp

```diff
@@ -0,0 +1,133 @@
+    // NOLINTNEXTLINE
+    StepSize = new ConfigDouble(cfg, "PathPlanner/RRT/StepSize", 0.15);
+    // NOLINTNEXTLINE
+    GoalBias = new ConfigDouble(
+        cfg, "PathPlanner/RRT/GoalBias", 0.3,
+        "Value from 0 to 1 that determines what proportion of the time the RRT "
+        "will grow towards the goal rather than towards a random point");
+    // NOLINTNEXTLINE
+    WaypointBias = new ConfigDouble(
+        cfg, "PathPlanner/RRT/WayPointBias", 0.5,
+        "Value from 0 to 1 that determines the portion of the time that the "
+        "RRT will"
+        " grow towards given waypoints rather than towards a random point");
+    // NOLINTNEXTLINE
+    MinIterations =
+        new ConfigInt(cfg, "PathPlanner/RRT/MinIterations", 100,
+                      "The minimum number of iterations for running RRT");
+    // todo(Ethan) can this be increased? RRT fails sometimes. testing needed
+    // //NOLINTNEXTLINE
+    MaxIterations =
+        new ConfigInt(cfg, "PathPlanner/RRT/MaxIterations", 250,
+                      "The maximum number of iterations for running RRT");
+}
+
+ConfigBool EnableExpensiveRRTDebugDrawing();
+
+void DrawRRT(const RRT::Tree<Point>& rrt, DebugDrawer* debug_drawer,
+             unsigned shellID) {
+    // Draw each robot's rrts in a different color
+    // Note: feel free to change these, they're completely arbitrary
+    static const std::array<QColor, 6> colors = {
+        QColor("green"), QColor("blue"),   QColor("yellow"),
+        QColor("red"),   QColor("purple"), QColor("orange")};
+    QColor color = colors[shellID % colors.size()];
+
+    for (const auto& node : rrt.allNodes()) {
+        if (node.parent() != nullptr) {
+            debug_drawer->drawLine(
+                Segment(node.state(), node.parent()->state()), color,
+                QString("RobotRRT%1").arg(shellID));
+        }
+    }
+}
+
+void DrawBiRRT(const RRT::BiRRT<Point>& biRRT, DebugDrawer* debug_drawer,
+               unsigned shellID) {
+    DrawRRT(biRRT.startTree(), debug_drawer, shellID);
+    DrawRRT(biRRT.goalTree(), debug_drawer, shellID);
+}
+
+vector<Point> runRRTHelper(Point start, Point goal, const ShapeSet& obstacles,
+                           const vector<Point>& waypoints, bool straightLine) {
+    auto state_space = std::make_shared<RoboCupStateSpace>(
+        Field_Dimensions::Current_Dimensions, obstacles);
+    RRT::BiRRT<Point> biRRT(state_space, Point::hash, 2);
+    biRRT.setStartState(start);
+    biRRT.setGoalState(goal);
+
+    if (straightLine) {
+        // Set the step size to be the distance between the start and goal.
+        biRRT.setStepSize(state_space->distance(start, goal));
+        // Plan straight toward the goal.
+        biRRT.setGoalBias(1);
+        // Try up to five times. If unsuccessful after five tries, there
+        // probably doesn't exist
+        // a straight path.
+        biRRT.setMinIterations(0);
+        biRRT.setMaxIterations(5);
```

Comment:
Why 5 times? Shouldn't it work the first time?

Reply:
No, it has a maximum travel distance.

This should probably be replaced with just a straight-line hit check, though.

Reply:
#1511

Reply:
I think it actually might since the step size is the distance between start and goal. I think we should still just swap to a straight line check since it's much more obvious what is happening

Reply:
Yeah, you right. Super weird - anyway, I'll leave it as a TODO.